classes/search/cms.php

```diff
@@ -0,0 +1,188 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Define search area.
+ *
+ * @package    mod_cms
+ * @author     Tomo Tsuyuki <tomotsuyuki@catalyst-au.com>
+ * @copyright  2024 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace mod_cms\search;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once($CFG->dirroot . '/mod/cms/lib.php');
+
+/**
+ * Define search area.
+ *
+ * @package    mod_cms
+ * @author     Tomo Tsuyuki <tomotsuyuki@catalyst-au.com>
+ * @copyright  2024 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class cms extends \core_search\base_mod {
```

Comment:
any reason for picking cms class  name? Looks like other activities have it as 'activity' e.g. mod_assign\search\activity

Reply:
I checked other activity modules, and they are using record name - e.g. `chapter`, `forum`, etc.
I renamed it to `cmsfield` as it's from cmsfield custom field handler.